subprojects/resources-s3/src/main/java/org/gradle/internal/resource/transport/aws/s3/S3Client.java

```diff
@@ -153,10 +154,31 @@ public S3Object getResource(URI uri) {
                     results.add(fileName);
                 }
             }
+
+        }
+
+        List<String> commonPrefixes = objectListing.getCommonPrefixes();
```

Comment:
Please extract a method for getting all the directory names from the objectListing as well as for the file names.